File: frontend/src/api/pptCache.ts
export interface DocumentCacheSummary {
  id: number;
  equipment_id: number;
  file_path: string;
  file_name: string;
  file_hash: string;
  modified_at: string;
  parsed_at: string;
  slide_count: number;
}

export interface SlideCacheItem {
  id: number;
  slide_number: number;
  title: string | null;
  content: string;
}

export interface DocumentCacheDetail extends DocumentCacheSummary {
  slides: SlideCacheItem[];
}

export interface PptAnalysisRequest {
  equipment_id: number;
  keywords: string[];
  date_from?: string;
  date_to?: string;
}

export interface SlideCandidateItem {
  document_cache_id: number;
  slide_cache_id: number;
  file_path: string;
  file_name: string;
  slide_number: number;
  title: string | null;
  content?: string | null;
  matched_keywords: string[];
  candidate_score: number;
  from_cache_search?: boolean;
}

export interface SourceFunctionItem {
  file_path: string | null;
  functions: string[];
  raw_text: string;
}

export interface ChangeItemCandidate {
  change_item_cache_id: number;
  document_cache_id: number;
  slide_no: number;
  file_path: string;
  file_name: string;
  item_no: string | null;
  change_title: string | null;
  csr_no: string | null;
  business_background: string | null;
  current_status: string | null;
  as_is: string | null;
  to_be: string | null;
  source_functions: SourceFunctionItem[];
  test_cases: string[];
  applicable_scopes: string[];
  matched_keywords: string[];
  candidate_score: number;
  from_cache_search?: boolean;
  from_fallback?: boolean;
  query_match_reasons?: Array<{
    keyword: string;
    field: string;
    value: string;
    score: number;
    strength?: string;
  }>;
  query_relevance_score?: number;
  query_relevance_level?: string;
}

export interface PptAnalysisResponse {
  equipment_id: number;
  ppt_candidate_count: number;
  processed_documents: number;
  cache_hits: number;
  cache_misses: number;
  parse_failures: number;
  documents: Array<{
    document_cache_id: number;
    file_path: string;
    file_name: string;
    slide_count: number;
    cache_hit: boolean;
  }>;
  slide_candidates: SlideCandidateItem[];
  change_item_candidates: ChangeItemCandidate[];
  fallback_documents_parsed: number;
  change_item_total: number;
  equipment_filter_excluded?: number;
}

const API_BASE = import.meta.env.VITE_API_BASE ?? "";

async function parseError(response: Response): Promise<string> {
  try {
    const data = await response.json();
    if (typeof data.detail === "string") return data.detail;
  } catch {
    /* ignore */
  }
  return `요청 실패 (${response.status})`;
}

export async function fetchPptCacheList(equipmentId: number): Promise<DocumentCacheSummary[]> {
  const response = await fetch(`${API_BASE}/api/equipment/${equipmentId}/ppt-cache`);
  if (!response.ok) throw new Error(await parseError(response));
  const data = await response.json();
  return data.documents;
}

export async function fetchPptCacheDetail(documentCacheId: number): Promise<DocumentCacheDetail> {
  const response = await fetch(`${API_BASE}/api/ppt-cache/${documentCacheId}`);
  if (!response.ok) throw new Error(await parseError(response));
  return response.json();
}

export async function deletePptCache(documentCacheId: number): Promise<void> {
  const response = await fetch(`${API_BASE}/api/ppt-cache/${documentCacheId}`, {
    method: "DELETE",
  });
  if (!response.ok) throw new Error(await parseError(response));
}

export async function runPptAnalysis(body: PptAnalysisRequest): Promise<PptAnalysisResponse> {
  const response = await fetch(`${API_BASE}/api/trace/ppt-analysis`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });
  if (!response.ok) throw new Error(await parseError(response));
  return response.json();
}

File: backend/app/api/ppt_cache.py
from fastapi import APIRouter, HTTPException

from app.schemas.ppt_cache import (
    DocumentCacheDetail,
    DocumentCacheListResponse,
    DocumentCacheSummary,
    SlideCacheItem,
)
from app.services.equipment_service import get_equipment
from app.services.ppt_cache_service import (
    delete_document_cache,
    get_document_cache_by_id,
    get_slides_for_document,
    list_document_cache_by_equipment,
)

router = APIRouter(prefix="/api", tags=["ppt-cache"])


@router.get(
    "/equipment/{equipment_id}/ppt-cache",
    response_model=DocumentCacheListResponse,
)
def list_ppt_cache(equipment_id: int) -> DocumentCacheListResponse:
    equipment = get_equipment(equipment_id)
    if equipment is None:
        raise HTTPException(status_code=404, detail="장비를 찾을 수 없습니다.")

    docs = list_document_cache_by_equipment(equipment_id)
    return DocumentCacheListResponse(
        equipment_id=equipment_id,
        documents=[
            DocumentCacheSummary(
                id=d.id,
                equipment_id=d.equipment_id,
                file_path=d.file_path,
                file_name=d.file_name,
                file_hash=d.file_hash,
                modified_at=d.modified_at,
                parsed_at=d.parsed_at,
                slide_count=d.slide_count,
            )
            for d in docs
        ],
    )


@router.get("/ppt-cache/{document_cache_id}", response_model=DocumentCacheDetail)
def get_ppt_cache_detail(document_cache_id: int) -> DocumentCacheDetail:
    doc = get_document_cache_by_id(document_cache_id)
    if doc is None:
        raise HTTPException(status_code=404, detail="PPT Cache를 찾을 수 없습니다.")

    slides = get_slides_for_document(document_cache_id)
    return DocumentCacheDetail(
        id=doc.id,
        equipment_id=doc.equipment_id,
        file_path=doc.file_path,
        file_name=doc.file_name,
        file_hash=doc.file_hash,
        modified_at=doc.modified_at,
        parsed_at=doc.parsed_at,
        slide_count=doc.slide_count,
        slides=[
            SlideCacheItem(
                id=s.id,
                slide_number=s.slide_number,
                title=s.title,
                content=s.content,
            )
            for s in slides
        ],
    )


@router.delete("/ppt-cache/{document_cache_id}", status_code=204)
def remove_ppt_cache(document_cache_id: int) -> None:
    doc = get_document_cache_by_id(document_cache_id)
    if doc is None:
        raise HTTPException(status_code=404, detail="PPT Cache를 찾을 수 없습니다.")
    delete_document_cache(document_cache_id)
